### PR TITLE
feat: Refactor and enhance Launch Configuration Logic for debugging generated apps in @sap-ux/launch-config

### PR DESCRIPTION
**Pull request by kjose90**
The implementation will focus on managing various workspace scenarios, including unsaved workspaces, folders without workspace files, and saved workspaces so users can seamlessly debug the generated application, regardless of their workspace configuration.

- [ ] `debug-config/workspaceManager.ts` 
Manages the configuration of the debug workspace based on the provided options.
It handles different scenarios such as whether a workspace is open, whether the project is inside or outside of a workspace, and other contextual factors.
`handleWorkspaceConfig` returns `WorkspaceHandlerInfo`, an object containing:
            1.  The path where the `launch.json` file will be written.
             2. The `cwd` command, determined by workspace settings.
             3. `workspaceFolderUri`, if provided, which facilitates workspace reloading.

- [ ] `debug-config/config.ts` 
Generates the contents of the `launch.json` file based on the options provided.
`configureLaunchJsonFile` returns the configured `launch.json `object, tailored to the specified debug options and cwd command.

- [ ] `launch-config-crud/create.ts`
Handles the creation of the launch configuration for the project based on the provided debug options.
`configureLaunchConfig` generates and creates the `launch.json` file.
If `workspaceFolderUri` is provided, it creates the `launch.json file`, triggers a workspace reload, and updates the `appInfo.json` file with the current project path. This ensures the App Info page is correctly loaded post-reload.

- [ ] Add unit tests

### CODE REVIEW

File: packages/launch-config/src/launch-config-crud/create.ts [hunk 1]
```diff
@@ -44,3 +51,127 @@ export async function createLaunchConfig(rootFolder: string, fioriOptions: Fiori
...
+ * @param {string} path - The project file path to add.
+ * @param log - The logger instance.
+ */
+export function writeApplicationInfoSettings(path: string, log?: Logger): void {
+    const appInfoFilePath: string = getFioriToolsDirectory();
+    console.log('appInfoFilePath', appInfoFilePath);
+    const appInfoContents = fs.existsSync(appInfoFilePath)
+        ? JSON.parse(fs.readFileSync(appInfoFilePath, 'utf-8'))
+        : { latestGeneratedFiles: [] };
+    console.log('appInfoContents', appInfoContents);
```
Comment (longieirl) [inline packages/launch-config/src/launch-config-crud/create.ts:68]: Will this print the JSON or just `object`?

Comment (kjose90) [reply packages/launch-config/src/launch-config-crud/create.ts]: Thanks for pointing this unnecessary console log - have removed this.

**Review by devinea — DISMISSED** [review]
Code changes are well structured, documented and tested. 
Changeset ✅

**Review by longieirl — APPROVED** [review]
Comments addressed.
Test coverage is good.
Changeset present.

**Comment by kjose90** [comment]
Thanks for reviewing 👍

File: packages/launch-config/package.json [hunk 2]
```diff
@@ -34,6 +34,9 @@
         "@sap-ux/project-access": "workspace:*",
         "@sap-ux/ui5-config": "workspace:*",
         "@sap-ux/ui5-info": "workspace:*",
+        "@sap-ux/odata-service-inquirer": "workspace:*",
```
Comment (IainSAP) [inline packages/launch-config/package.json:37]: Please remove this dependency, this is using types not intended for this purpose and should only be consumed if you are calling `getPrompts` or `prompt`

**Pull request merged**